sdk/metric/sdk.go

```diff
@@ -419,10 +418,8 @@ func (m *Accumulator) observeAsyncInstruments(ctx context.Context) int {
 	defer m.asyncLock.Unlock()
 
 	asyncCollected := 0
-	m.asyncContext = ctx
 
-	m.asyncInstruments.Run(context.Background(), m)
```

Comment:
This is a change in behavior (a bugfix?), ideally should be done in a separate PR.

Reply:
Done, thanks.